cmd/apichecker/apicheck.go

```diff
@@ -0,0 +1,1226 @@
+		configData.CardID = trelloCardID
+		configData.ChecklistID = trelloChecklistID
+		err = CheckUpdates(jsonFile, &configData)
+		if err != nil {
+			log.Fatal(err)
+		}
+	} else {
+		log.Printf("This is a test update since API keys are not set.\n")
+		err := CheckUpdates(testJSONFile, &testConfigData)
+		if err != nil {
+			log.Fatal(err)
+		}
+	}
+}
+
+// CanUpdateTrello checks if all the data necessary for updating trello is available
+func CanUpdateTrello() bool {
+	if AreAPIKeysSet() && IsTrelloBoardDataSet() {
+		return true
+	}
+	return false
+}
+
+// AreAPIKeysSet checks if api keys and tokens are set
+func AreAPIKeysSet() bool {
+	if (apiKey != "" && apiToken != "") || (configData.Key != "" && configData.Token != "") {
+		return true
+	}
+	return false
+}
+
+// IsTrelloBoardDataSet checks if data required to update trello board is set
+func IsTrelloBoardDataSet() bool {
+	if (trelloBoardID != "" && trelloListID != "" && trelloChecklistID != "" && trelloCardID != "") ||
+		(configData.CardID != "" && configData.ChecklistID != "") {
+		return true
+	}
+	return false
+}
+
+// getSha gets the sha of the latest commit
+func getSha(repoPath string) (ShaResponse, error) {
+	var resp ShaResponse
+	path := fmt.Sprintf(githubPath, repoPath)
+	if verbose {
+		fmt.Printf("Getting SHA of this path: %v\n", path)
+	}
+	return resp, common.SendHTTPGetRequest(path, true, verbose, &resp)
+}
+
+// CheckExistingExchanges checks if the given exchange exists
+func CheckExistingExchanges(exchName string, confData *Config) bool {
+	for x := range confData.Exchanges {
+		if confData.Exchanges[x].Name == exchName {
+			return true
+		}
+	}
+	return false
+}
+
+// CheckMissingExchanges checks if any supported exchanges are missing api checker functionality
+func CheckMissingExchanges(confData *Config) []string {
+	var tempArray []string
+	for x := range confData.Exchanges {
+		tempArray = append(tempArray, confData.Exchanges[x].Name)
+	}
+	supportedExchs := exchange.Exchanges
+	for z := 0; z < len(supportedExchs); {
+		if common.StringDataContainsInsensitive(tempArray, supportedExchs[z]) {
+			supportedExchs = append(supportedExchs[:z], supportedExchs[z+1:]...)
+			continue
+		}
+		z++
+	}
+	return supportedExchs
+}
+
+// ReadFileData reads the file data from the given json file
+func ReadFileData(fileName string) (Config, error) {
+	var c Config
+	file, err := os.Open(fileName)
+	if err != nil {
+		return c, err
+	}
+	defer file.Close()
+	byteValue, err := ioutil.ReadAll(file)
+	if err != nil {
+		return c, err
+	}
+	err = json.Unmarshal(byteValue, &c)
+	if err != nil {
+		return c, err
+	}
+	return c, nil
+}
+
+// CheckUpdates checks updates.json for all the existing exchanges
+func CheckUpdates(fileName string, confData *Config) error {
+	var resp []string
+	errMap := make(map[string]error)
+	var wg sync.WaitGroup
+	var m sync.Mutex
+	allExchangeData := confData.Exchanges
+	for x := range allExchangeData {
+		wg.Add(1)
+		go func(x int) {
```

Comment:
Is there much of a need to make this asynchronous if you're locking it in both cases?